helios-client/src/main/java/com/spotify/helios/client/AuthenticatingHttpConnector.java

```diff
@@ -105,8 +111,11 @@ public HttpURLConnection connect(final URI uri, final String method, final byte[
     try {
       log.debug("connecting to {}", ipUri);
 
-      if (clientCertificatePath.isPresent()) {
-        // prioritize using the certificate file if set
+      if (accessTokenOpt.isPresent()) {
+        // prioritize using bearer token
+        return connectWithBearerToken(ipUri, method, entity, headers);
+      } else if (clientCertificatePath.isPresent()) {
```

Comment:
Why not send the header along with the certificate if both are present, rather than making this an either-or? Seems like they could be separate concerns as one happens at the SSL layer and the other at the HTTP request layer.